cmd/agent/common/common.go

```diff
@@ -82,10 +83,49 @@ func NewRemoteConfigService(hostname string) (*remoteconfig.Service, error) {
 	baseRawURL := configUtils.GetMainEndpoint(config.Datadog, "https://config.", "remote_configuration.rc_dd_url")
 	traceAgentEnv := configUtils.GetTraceAgentDefaultEnv(config.Datadog)
 
-	configService, err := remoteconfig.NewService(config.Datadog, apiKey, baseRawURL, hostname, remoteconfig.WithTraceAgentEnv(traceAgentEnv))
+	telemetryReporter := newRcTelemetryReporter()
+
+	configService, err := remoteconfig.NewService(config.Datadog, apiKey, baseRawURL, hostname, telemetryReporter, remoteconfig.WithTraceAgentEnv(traceAgentEnv))
 	if err != nil {
 		return nil, fmt.Errorf("unable to create remote-config service: %w", err)
 	}
 
 	return configService, nil
 }
+
+// ddRcTelemetryReporter is a datadog-agent telemetry counter for RC cache bypass metrics. It implements the RcTelemetryReporter interface.
+type ddRcTelemetryReporter struct {
+	BypassRateLimitCounter telemetry.Counter
+	BypassTimeoutCounter   telemetry.Counter
+}
+
+// IncRateLimit increments the ddRcTelemetryReporter BypassRateLimitCounter counter.
+func (r *ddRcTelemetryReporter) IncRateLimit() {
+	r.BypassRateLimitCounter.Inc()
+}
+
+// IncTimeout increments the ddRcTelemetryReporter BypassTimeoutCounter counter.
+func (r *ddRcTelemetryReporter) IncTimeout() {
+	r.BypassTimeoutCounter.Inc()
+}
+
+// newRcTelemetryReporter returns a new ddRcTelemetryReporter that uses the datadog-agent telemetry package to emit metrics.
+func newRcTelemetryReporter() *ddRcTelemetryReporter {
+	commonOpts := telemetry.Options{NoDoubleUnderscoreSep: true}
+	return &ddRcTelemetryReporter{
+		BypassRateLimitCounter: telemetry.NewCounterWithOpts(
+			"remoteconfig",
+			"cache_bypass_ratelimiter_skip",
+			[]string{},
+			"Number of Remote Configuration cache bypass requests skipped by rate limiting.",
+			commonOpts,
+		),
+		BypassTimeoutCounter: telemetry.NewCounterWithOpts(
+			"remoteconfig",
+			"cache_bypass_timeout",
+			[]string{},
+			"Number of Remote Configuration cache bypass requests that timeout.",
+			commonOpts,
+		),
+	}
+}
```

Comment:
This feels out of place in `cmd/agent/common/common.go`. The remote-config telemetry should be contained in the remote-config package/comp. Why was it moved here ?

Reply:
The underlying reason it's here now is because the remote config service is not modularized and is being instantiated in core and cluster agents today. It hasn't been rewritten in a way that it can be supplied via uber.fx the way the rest of the `comp` components are, and in order to untangle the dependencies described in the PR description it can't live in `pkg/config/remote` where it is being abstracted from in this PR.

I think the right way to do this is to uber.fx-ify the remote config service and supply it from `comp`, but I was trying to keep the PR slim and have that effort be addressed separately. In the meantime, would it be fair to move this to a new file owned by RC ([as suggested by](https://github.com/DataDog/datadog-agent/pull/21508#discussion_r1425481332) @pgimalac ) with a TODO to use uber.fx in the longer-term?

Reply:
Thanks for the explanation. A dedicated file sounds like a good compromise as long as we have a card to track the remaining work.

Reply:
Sorry for the delay on these changes! I've pulled all the remote config service initialization to [its own common file with a corresponding TODO](https://github.com/DataDog/datadog-agent/pull/21508/files#diff-5c0e5dc3a96087fab708008961fe45bdc55f5d9a16393e31c23294b22482740fR5) linking the ticket to do the larger refactor